### PR TITLE
Setup JavaScript linting

### PR DESCRIPTION
**Pull request by alex-ju**
## What
- Setup JavaScript linting using StandardJS (by GDS convention) and Yarn (by GOV.UK convention).
- Autofix linting issues using `standard --fix`
- Manually fix linting issues that were not solved by the automated task.
- Update rake task to run all linters (Rubocop and Standard, for now) and setup CI (GitHub and Jenkins) to run it
- Update documentation on dependencies

## Why
For code consistency, improved maintainability, and increasing code quality (especially on this repo that in the current context tends to accumulate tech debt quite quickly).

## Context
This setup is similar to the one in [Content Publisher](https://github.com/alphagov/content-publisher/pull/93), [GOV.UK Publishing Components](https://github.com/alphagov/govuk_publishing_components/pull/758) and, more recently, [Whitehall](https://github.com/alphagov/whitehall/pull/5848).
A separate PR will follow to add SCSS linting to this repo.

## Local setup
[Add standard to your favorite code editor](https://standardjs.com/#are-there-text-editor-plugins) if you write JavaScript and haven't done it already.

### CODE REVIEW

**Comment by benthorner** [comment]
Removed CD warning.

We're disabling automatic deployments for this app until it meets the safety criteria identified by RFC 128.

https://github.com/alphagov/collections/pull/2059

File: README.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ Content for taxon pages is returned by a search in Rummager based on content_ids
     - content tagged to a particular Topic, Mainstream browse page or Organisation
 - [email-alert-api](https://github.com/alphagov/email-alert-api), provides:
     - support for subscribing to notifications from a topic
+- [yarn][] - package manager for JavaScripts
```
Comment (owenatgov) [inline README.md:115]: Should this link to [this](https://yarnpkg.com/)? Or even be a link?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "collections",
+  "description": "Serves GOV.UK navigation pages, browse, topic, step-by-steps & services and information pages.",
+  "private": true,
```
Comment (owenatgov) [inline package.json:4]: I guess there's no harm in this attribute being here but is this appropriate as this repo itself is public? Sorry if I've misunderstood the intent here.

**Review by owenatgov — COMMENTED** [review]
Looking great. Just a couple of pedantic comments.

File: README.md (hunk 1 above)
Comment (alex-ju) [reply README.md:115]: Ah, well it's using the reference style (with a [definition at the end of the document](https://github.com/alphagov/collections/pull/2070/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R156)). I often prefer this way as it's less verbose and if you need to update the link you only do it once per markdown page. It [renders fine](https://github.com/alphagov/collections/pull/2070/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5), but equally happy to make it a link.

File: package.json (hunk 2 above)
Comment (alex-ju) [reply package.json:4]: By setting it to private we express the fact that this project is not meant to be published via npm (npm checks for this value before allowing code to be published).

**Comment by alex-ju** [comment]
Rebased to master.

File: README.md (hunk 1 above)
Comment (owenatgov) [reply README.md:115]: It wasn't working when I looked at it before but it looks fine now so I'll withdraw this!

Comment (alex-ju) [reply README.md:115]: I initial forgot a space before the definitions at the end. So your comment helped me spot that!

**Comment by alex-ju** [comment]
Closing this as it stalled in review – maybe we'll get more luck with the upcoming https://github.com/alphagov/collections/tree/node-linters by @kevindew.

**Comment by kevindew** [comment]
Thanks @alex-ju - it's all ready for review https://github.com/alphagov/collections/pull/2121

**Pull request closed without merging**